src/main/java/org/folio/innreach/domain/service/impl/RecordContributionServiceImpl.java

```diff
@@ -109,22 +116,47 @@ private void logItemTransformationError(Item item, Exception e) {
   }
 
   private InnReachResponse contributeBib(UUID centralServerId, String bibId, BibInfo bib) {
+    log.info("Retry happening for contributeBib with bibId: {}",bibId);
     var response = irContributionService.contributeBib(centralServerId, bibId, bib);
+    checkServiceSuspension(response);
     Assert.isTrue(response.isOk(), "Unexpected contribution response: " + response);
     return response;
   }
 
   private InnReachResponse verifyBibContribution(UUID centralServerId, String bibId) {
+    log.info("verifyBibContribution with bibId: {}",bibId);
     var response = irContributionService.lookUpBib(centralServerId, bibId);
+    checkServiceSuspension(response);
     Assert.isTrue(response.isOk(), "Unexpected verification response: " + response);
     return response;
   }
 
   private InnReachResponse contributeBibItems(String bibId, UUID centralServerId, List<BibItem> bibItems) {
     var response = irContributionService.contributeBibItems(centralServerId, bibId, BibItemsInfo.of(bibItems));
+    checkServiceSuspension(response);
     Assert.isTrue(response.isOk(), "Unexpected items contribution response: " + response);
     return response;
   }
 
-
+  private void checkServiceSuspension(InnReachResponse response) {
+    if (response!=null && response.getErrors()!=null && !response.getErrors().isEmpty()) {
+      InnReachResponse.Error errorResponse = response.getErrors().get(0);
+
+      var error = errorResponse!=null ? errorResponse.getReason() : "";
+      String errorMessages = "";
+
+      if(errorResponse!=null && errorResponse.getMessages()!=null && !errorResponse.getMessages().isEmpty()) {
+        errorMessages = errorResponse.getMessages().get(0);
+      }
+      log.info("checkServiceSuspension:: error is : {}",error);
+      if (error.contains(CONTRIBUTION_TO_D2IRM_IS_CURRENTLY_SUSPENDED)) {
+        log.info("ServiceSuspendedException occur");
+        throw new ServiceSuspendedException(CONTRIBUTION_TO_D2IRM_IS_CURRENTLY_SUSPENDED);
+      }
+      if(errorMessages.contains(CONNECTIONS_ALLOWED_FROM_THIS_SERVER)) {
+        log.info("InnReachConnectionException occur");
```

Comment:
This ifo log message here and the one above should not be the same. Otherwise we won't be able to tell in logging which one happened.